googletest/src/gtest.cc

```diff
@@ -5385,4 +5385,23 @@ void InitGoogleTest(int* argc, wchar_t** argv) {
 #endif  // defined(GTEST_CUSTOM_INIT_GOOGLE_TEST_FUNCTION_)
 }
 
+std::string TempDir() {
```

Comment:
The implementation of TempDir is very related to portability so I think it should remain there as-is.
This change might just expose a forwarding function called testing::TempDir() that calls it?